crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Generic.hs

```diff
@@ -643,16 +645,28 @@ readMem sym w l tp alignment m = do
   sz         <- bvLit sym w (bytesToInteger (typeEnd 0 tp))
   p1         <- isAllocated sym w alignment l (Just sz) m
   p2         <- isAligned sym w l alignment
-  readMem' sym w (memEndianForm m) l tp alignment (memWrites m) >>=
-    \case
-      e@(W4P.Err _) -> return e
-      PartLLVMVal p v ->
-        let ub1 = UB.ReadUnallocated  (UB.pointerView l)
-            ub2 = UB.ReadBadAlignment (UB.pointerView l) alignment
-            p'  = W4AT.And (p :| [ W4AT.Leaf (undefinedBehavior ub1 p1)
-                                 , W4AT.Leaf (undefinedBehavior ub2 p2)
-                                 ])
-        in return $ PartLLVMVal p' v
+  maybe_allocation_array <- asMemAllocationArrayStore sym w l m
+  part_val <- case maybe_allocation_array of
+    Just (arr, _arr_sz) -> do
```

Comment:
Is is correct to be ignoring the array size here? I don't recall exactly where we check that reads are in bounds, but this looks suspicious.

Reply:
it is safe to ignore the array size, this code runs after the bounds checks.